runChecks.py

```diff
@@ -0,0 +1,134 @@
+        { "pattern" : '<Eigen', "message" : 'File includes an Eigen header.'},
+        { "pattern" : 'Eigen::', "message" : 'File uses Eigen.'}
+    ]
+    errors += grep_patterns('scal', 'stan/math/*/scal', scal_checks)
+
+    # Check for files inside stan/math/*/arr that contain stan/math/*/mat or Eigen
+    arr_checks = [
+        { "pattern" : '<stan/math/.*/mat/', "message" : 'File includes an matrix header file.'},
+        { "pattern" : '<Eigen', "message" : 'File includes an Eigen header.'},
+        { "pattern" : 'Eigen::', "message" : 'File uses Eigen.'}
+    ]    
+    errors += grep_patterns('arr', 'stan/math/*/arr', arr_checks)
+
+    # Check to make sure we use C++14 constructs in stan/math
+    cpp14_checks = [
+        { "pattern" : 'boost::is_unsigned<', "message" : 'File uses boost::is_unsigned instead of std::is_unsigned.'},
+        { "pattern" : '<boost/type_traits/is_unsigned>',\
+            "message" : 'File includes <boost/type_traits/is_unsigned.hpp> instead of <type_traits>.'},
+        { "pattern" : 'boost::is_arithmetic<', "message" : 'File uses boost::is_arithmetic instead of std::is_arithmetic.'},
+        { "pattern" : '<boost/type_traits/is_arithmetic.hpp>',\
+            "message" : 'File includes <boost/type_traits/is_unsigned.hpp> instead of <type_traits>.'},
+        { "pattern" : 'boost::is_convertible<', "message" : 'File uses boost::is_convertible instead of std::is_convertible.'},
+        { "pattern" : '<boost/type_traits/is_convertible.hpp>',\
+            "message" : 'File includes <boost/type_traits/is_convertible.hpp> instead of <type_traits>.'},
+        { "pattern" : 'boost::is_same<', "message" : 'File uses boost::is_same instead of std::is_same.'},
+        { "pattern" : '<boost/type_traits/is_same.hpp>',\
+            "message" : 'File includes <boost/type_traits/is_same.hpp> instead of <type_traits>.'},
+        { "pattern" : 'boost::enable_if_c<', "message" : 'File uses boost::enable_if_c instead of std::enable_if.'},
+        { "pattern" : 'boost::enable_if<', "message" : 'File uses boost::enable_if instead of std::enable_if.'},
+        { "pattern" : 'boost::disable_if<', "message" : 'File uses boost::disable_if instead of std::enable_if.'},
+        { "pattern" : '<boost/utility/enable_if.hpp>',\
+            "message" : 'Replace \<boost/utility/enable_if.hpp\> with \<type_traits\>.'}
+    ]    
+    errors += grep_patterns('C++14', 'stan/math', cpp14_checks)
+    
+    # Check for includes of stan/math/*/meta/*.hpp inside stan/math, excluding meta.hpp files and the /meta subfolder
+    meta_checks = [
+        { "pattern" : '<stan/math/.*/meta/.*hpp', "message" : 'File includes */meta/*.hpp header file. Should include meta.hpp'}
+    ]
+    meta_exclude = ['stan/math/*/*/meta', 'stan/math/*/meta.hpp']
+    errors += grep_patterns('meta', 'stan/math', meta_checks, meta_exclude)
+
+    errors += check_non_test_files_in_test()
```

Comment:
`errors += check_non_test_files_in_test()`

Reply:
You probably meant use .extend here. Thank you.